app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/HomeScreenShimmerLoading.kt

```diff
@@ -0,0 +1,134 @@
+package com.kks.nimblesurveyjetpackcompose.ui.presentation.common
+
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import androidx.compose.animation.core.*
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+
+@Composable
+fun HomeScreenShimmerLoading() {
+    val shimmerColors = listOf(
+        Color.LightGray.copy(alpha = 0.4f),
+        Color.LightGray.copy(alpha = 0.2f),
+        Color.LightGray.copy(alpha = 0.4f),
+    )
+
+    val transition = rememberInfiniteTransition()
+    val translateAnimation = transition.animateFloat(
+        initialValue = 0f,
+        targetValue = 1000f,
+        animationSpec = infiniteRepeatable(
+            animation = tween(
+                durationMillis = 1000,
+                easing = FastOutSlowInEasing
+            ),
+            repeatMode = RepeatMode.Reverse
+        )
+    )
+
+    val brush = Brush.linearGradient(
+        colors = shimmerColors,
+        start = Offset.Zero,
+        end = Offset(x = translateAnimation.value, y = translateAnimation.value)
+    )
```

Comment:
we could use a `horizontalGradient` here instead 😄 
```suggestion
    return horizontalGradient(
        colors = shimmerColors,
        endX = translateAnimation.value
    )
```

Reply:
Since current implementation also works well, I will leave it like that 🙏